### PR TITLE
Add Character Prefixes and Suffixes to Text Events 

### PR DESCRIPTION
**Pull request by CakeVR**
This new Feature allows to set text segments before and after a text spoken by a character.

For instance, you can set the `“` or `"` for your quotation marks or use rich tags, such as colour.

### CODE REVIEW

File: addons/dialogic/Editor/CharacterEditor/character_prefix_suffix.tscn [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+layout_mode = 2
+tooltip_text = "Choices can appear either instantly when the text finished, after a delay, a click or either."
+texture = null
+hint_text = "If a character speaks, this appears before their text.
+Example: Colour Tags or Quotation Marks."
+
+[node name="PrefixInput" type="LineEdit" parent="."]
+layout_mode = 2
+size_flags_horizontal = 3
+tooltip_text = "Mood name"
```
Comment (salianifo) [inline addons/dialogic/Editor/CharacterEditor/character_prefix_suffix.tscn:31]:
```suggestion
```

File: addons/dialogic/Editor/CharacterEditor/character_prefix_suffix.tscn [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+tooltip_text = "If a character speaks, this appears before their text.
+Example: Colour Tags or Quotation Marks."
+texture = null
+hint_text = "If a character speaks, this appears after their text.
+Example: Colour Tags or Quotation Marks."
+
+[node name="SuffixInput" type="LineEdit" parent="."]
+layout_mode = 2
+size_flags_horizontal = 3
+tooltip_text = "Mood name"
```
Comment (salianifo) [inline addons/dialogic/Editor/CharacterEditor/character_prefix_suffix.tscn:53]:
```suggestion
```

**Review by salianifo — CHANGES_REQUESTED** [review]
This is such a handy feature!

**Comment by Jowan-Spooner** [comment]
I very much like this. 
I'd remove the placeholder quotation marks in the line edits, because with just a symbol it's a bit unclear that it's a placeholder.

I'd also recommend doing it like this:
```gdscript
	if current_character:
		var character_prefix: String = current_character.custom_info.get(DialogicCharacterPrefixSuffixSection.PREFIX_CUSTOM_KEY, DialogicCharacterPrefixSuffixSection.DEFAULT_PREFIX)
		var character_suffix: String = current_character.custom_info.get(DialogicCharacterPrefixSuffixSection.SUFFIX_CUSTOM_KEY, DialogicCharacterPrefixSuffixSection.DEFAULT_SUFFIX)
		text = character_prefix + text + character_suffix
					
	text_node.reveal_text(text, additional)
```
Because this way the modified version is what is getting returned by `Dialogic.Text.update_dialog_text()` and thus also what is stored in the simple history.

Or is there a reason you didn't do it that way?

**Comment by CakeVR** [comment]
> I very much like this. I'd remove the placeholder quotation marks in the line edits, because with just a symbol it's a bit unclear that it's a placeholder.
> 
> I'd also recommend doing it like this:
> 
> ```gdscript
> 	if current_character:
> 		var character_prefix: String = current_character.custom_info.get(DialogicCharacterPrefixSuffixSection.PREFIX_CUSTOM_KEY, DialogicCharacterPrefixSuffixSection.DEFAULT_PREFIX)
> 		var character_suffix: String = current_character.custom_info.get(DialogicCharacterPrefixSuffixSection.SUFFIX_CUSTOM_KEY, DialogicCharacterPrefixSuffixSection.DEFAULT_SUFFIX)
> 		text = character_prefix + text + character_suffix
> 					
> 	text_node.reveal_text(text, additional)
> ```
> 
> Because this way the modified version is what is getting returned by `Dialogic.Text.update_dialog_text()` and thus also what is stored in the simple history.
> 
> Or is there a reason you didn't do it that way?

Thanks! I have applied your suggestions.

**Comment by Jowan-Spooner** [comment]
Cool. Looks good to me now!

**Pull request merged**